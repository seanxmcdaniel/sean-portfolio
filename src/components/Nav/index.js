import React, {useEffect} from 'react';
import { capitalizeFirstLetter } from '../../utils/helpers'

function Nav(props) {

  const {
    categories = [],
    setCurrentCategory,
    currentCategory,
    setContactSelected
  } = props;

  return (
    <header className="flex-row px-1">
      <h2 className='headername'>
        <a data-testid="link" href="/">
          Sean McDaniel
        </a>
      </h2>
      <nav>
        <ul className="flex-row">
          <li className="mx-2">
            <span onClick={() => {setContactSelected(false); setCurrentCategory(false)}}>
              About Me
            </span>
          </li>
          <li className={"mx-2"}>
            <span onClick={() => setContactSelected(true)}>
              Contact
            </span>
          </li>
          {categories.map((category) => (
            <li
              className={`mx-1 ${
                currentCategory.name === category.name
                }`}
              key={category.name}
            >
              <span
                onClick={() => {
                  setCurrentCategory(category);
                  setContactSelected(false);
                }}
              >
                {capitalizeFirstLetter(category.name)}
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}
  
  export default Nav;