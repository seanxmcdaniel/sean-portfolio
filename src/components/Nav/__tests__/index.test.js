import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
    //Baseline test
    it('renders', () => {
        render(<Nav />)
    })

    //Snapshot test
    it('Matches snapshot DOM node structures', () => {
        const { asFragment } = render(<Nav />);
        expect(asFragment()).toMatchSnapshot();
    })
})