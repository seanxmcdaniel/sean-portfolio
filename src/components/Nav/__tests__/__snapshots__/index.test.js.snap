// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component Matches snapshot DOM node structures 1`] = `
<DocumentFragment>
  <header
    class="flex-row"
  >
    <h2>
      <a
        href="/"
      >
        Sean McDaniel
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            href="#about"
          >
            About Me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Projects
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
